### PR TITLE
[Translator] Rename namespace to Azure.AI.Translation.Document

### PR DESCRIPTION
**Pull request by maririos**
After more architects and service discussion, name for namespace should be `Azure.AI.Translation.Document`.

This PR updates all code. A future PR will rename the `documentranslation` folder to `translation` and create the respective pipelines

### CODE REVIEW

**Comment by maririos** [comment]
FYI All live test failures are service issues. I already opened bugs for each failure against the service.

**Review by kristapratico — APPROVED** [review]
all instances of rename LGTM

File: sdk/documenttranslation/Azure.AI.Translation.Document/Azure.AI.Translation.Document.sln
Comment (heaths) [inline sdk/documenttranslation/Azure.AI.Translation.Document/Azure.AI.Translation.Document.sln:6]: This doesn't follow naming with other packages. I realize this is the latest discussion, but I've replied to the email. Following suite with other libraries, this should be `Azure.AI.Translation.Documents`.

Comment (maririos) [reply sdk/documenttranslation/Azure.AI.Translation.Document/Azure.AI.Translation.Document.sln:6]: It is an in-between and is prioritizing service consistency (Text and Document singular). 
Let me know if you consider this a blocker

Comment (heaths) [reply sdk/documenttranslation/Azure.AI.Translation.Document/Azure.AI.Translation.Document.sln:6]: Not a blocker, but with the name switch-around from "Document Translator" (or even "Document Translation") to "Translation.Document", the product name is lost. These packages and namespaces are conceptual. Given the guidelines, if anything the service name is in the second or third position (most often the third, as the second position is a topic, like "AI" or "Security".

This is no different than Blob Storage becoming Storage.Blobs.

/cc @johanste @KrzysztofCwalina

Comment (heaths) [reply sdk/documenttranslation/Azure.AI.Translation.Document/Azure.AI.Translation.Document.sln:6]: That said, this is not a problem to be solved and hold up this PR, so I'll sign off.

**Review by heaths — APPROVED** [review]

File: sdk/documenttranslation/Azure.AI.Translation.Document/Azure.AI.Translation.Document.sln
Comment (maririos) [reply sdk/documenttranslation/Azure.AI.Translation.Document/Azure.AI.Translation.Document.sln:6]: Although it is true the service name is `Translator`, `Translation` is the user-friendly word the service has been using for comunication. 
Also, service has given us green light with this name.

For the upcoming release of `1.0.0-beta.1` we are going to go with `Azure.AI.Translation.Document` if, after more conversations, the name needs to change, although not ideal, we can definitely do it.

**Pull request merged**
